### PR TITLE
Migrate VSTS YAML to dev15.7.x

### PR DESCRIPTION
**Pull request by jaredpar**
This copies our VSTS YAML file into dev15.7.x. This changes nothing
about our official build in this branch. It just seeds it with the
correct value and lets the merge bots move it to the rest of the trees.

The official switch over for 15.7 will happen at a later time and will
be very deliberate.

<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes

(either VSO or GitHub links)

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

File: build/Targets/Packages.props [hunk 1]
```diff
@@ -152,7 +152,7 @@
     <RoslynToolsMicrosoftVSIXExpInstallerVersion>0.4.0-beta</RoslynToolsMicrosoftVSIXExpInstallerVersion>
     <RoslynToolsMSBuildVersion>0.4.0-alpha</RoslynToolsMSBuildVersion>
     <RoslynToolsReferenceAssembliesVersion>0.1.3</RoslynToolsReferenceAssembliesVersion>
-    <RoslynToolsSignToolVersion>1.0.0-beta2-dev2</RoslynToolsSignToolVersion>
+    <RoslynToolsSignToolVersion>1.0.0-beta2-dev3</RoslynToolsSignToolVersion>
```
Comment (jaredpar) [inline build/Targets/Packages.props]: This just moves the sign tool to a version that can properly do test signing.

File: .vsts-ci.yml [hunk 2]
```diff
@@ -0,0 +1,149 @@
+resources:
+- repo: self
```
Comment (agocke) [inline .vsts-ci.yml]: How was this file generated?

Comment (jaredpar) [reply .vsts-ci.yml]: This is generated initially by using the "View YAML" link on our old VSTS build definition. It's been altered a bit from that initial definition though.

**Comment by agocke** [comment]
LGTM (belated)

**Pull request merged**